chain/store/store.go

```diff
@@ -378,19 +378,30 @@ func (cs *ChainStore) SetGenesis(ctx context.Context, b *types.BlockHeader) erro
 }
 
 func (cs *ChainStore) PutTipSet(ctx context.Context, ts *types.TipSet) error {
+	for _, blk := range ts.Blocks() {
+		if err := cs.AddToTipSetTracker(ctx, blk); err != nil {
+			return xerrors.Errorf("failed to add block to ts tracker: %w", err)
+		}
+	}
+
 	if err := cs.PersistTipsets(ctx, []*types.TipSet{ts}); err != nil {
 		return xerrors.Errorf("failed to persist tipset: %w", err)
 	}
 
-	expanded, err := cs.expandTipset(ctx, ts.Blocks()[0])
+	expanded, err := cs.ExpandTipsetByHeightAndParent(ctx, ts.Height(), ts.Parents().Cids())
```

Comment:
I'm tempted to just use the new `FormBestTipSetForHeight` method here as well, and drop `ExpandTipsetByHeightAndParent` entirely.

Reply:
It's a bit slower, but I think I'd prefer doing it anyway